geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/partitions/DateTimeScheme.scala

```diff
@@ -172,20 +174,37 @@ object DateTimeScheme {
 
   object Formats {
 
-    private val all = Seq(Minute, Hourly, Daily, Weekly, Monthly, JulianMinute, JulianHourly, JulianDaily)
-
     def apply(name: String): Option[Format] = all.find(_.name.equalsIgnoreCase(name))
 
-    sealed case class Format private [Formats] (name: String, format: String, unit: ChronoUnit)
+    case class Format private[Formats] (name: String, formatter: DateTimeFormatter, unit: ChronoUnit)
+
+    private[Formats] object Format {
+      def apply(name: String, format: String, unit: ChronoUnit): Format =
+        Format(name, DateTimeFormatter.ofPattern(format), unit)
+    }
 
-    object Minute       extends Format("minute",        "yyyy/MM/dd/HH/mm", ChronoUnit.MINUTES)
-    object Hourly       extends Format("hourly",        "yyyy/MM/dd/HH",    ChronoUnit.HOURS  )
-    object Daily        extends Format("daily",         "yyyy/MM/dd",       ChronoUnit.DAYS   )
-    object Weekly       extends Format("weekly",        "yyyy/ww",          ChronoUnit.WEEKS  )
-    object Monthly      extends Format("monthly",       "yyyy/MM",          ChronoUnit.MONTHS )
-    object JulianMinute extends Format("julian-minute", "yyyy/DDD/HH/mm",   ChronoUnit.MINUTES)
-    object JulianHourly extends Format("julian-hourly", "yyyy/DDD/HH",      ChronoUnit.HOURS  )
-    object JulianDaily  extends Format("julian-daily",  "yyyy/DDD",         ChronoUnit.DAYS   )
+    val Minute       : Format = Format("minute",        "yyyy/MM/dd/HH/mm", ChronoUnit.MINUTES)
```

Comment:
Would changing from an object to val cause a small, external "api" change?

Reply:
possibly? i'm not sure if there are any actual differences between a val and an object in this case. but i don't think this qualifies as part of the public api - the configuration is still the same. if you object, I can change it back, it just seemed a little odd that they were objects.

Reply:
We can leave it.  I'm on a streak where I try to think through when an API _may_ be changing.  :)